src/fetch/ipfs.py

```diff
@@ -62,11 +66,13 @@ def get_content(self, max_retries: int = 3) -> Optional[Any]:
 
     @classmethod
     async def fetch_many(  # pylint: disable=too-many-locals
-        cls, missing_rows: list[dict[str, str]], max_retries: int = 3
+        cls, missing_rows: list[dict[str, str]], access_token: str, max_retries: int = 3
     ) -> tuple[list[FoundContent], list[NotFoundContent]]:
         """Async AppData Fetching"""
         found, not_found = [], []
-        async with aiohttp.ClientSession() as session:
+        async with aiohttp.ClientSession(
+            headers={"x-pinata-gateway-token": access_token}
```

Comment:
nit: move to a const to keep it DRY (line 57 also sets that header

Reply:
Also, is the header needed in both the request, and this ClientSession?

Reply:
Yes -- two totally different requests (one is an async library and the other is not).